client/mm/exchange_adaptor.go

```diff
@@ -2428,39 +2384,19 @@ func (u *unifiedExchangeAdaptor) checkPendingDEXOrderTxs(pendingOrder *pendingDE
 	// }
 
 	for _, tx := range pendingOrder.redeems {
-		isDynamicSwapper := u.isDynamicSwapper(toAsset)
-
-		// For dynamic fee assets, the fees are paid from the active balance,
-		// and are not taken out of the redeem amount.
-		if isDynamicSwapper || tx.Confirmed {
-			effects.availableDiff[toFeeAsset] -= int64(tx.Fees)
-		} else {
-			effects.pending[toFeeAsset] -= tx.Fees
-		}
-		addFees(false, tx.Fees)
-		addDelta(false, int64(tx.Amount))
+		effects.fees[toFeeAsset] += tx.Fees
 
 		if tx.Confirmed {
-			effects.availableDiff[toAsset] += int64(tx.Amount)
+			effects.settled[toAsset] += int64(tx.Amount)
 		} else {
 			effects.pending[toAsset] += tx.Amount
 		}
 	}
 	for _, tx := range pendingOrder.refunds {
-		isDynamicSwapper := u.isDynamicSwapper(fromAsset)
-
-		// For dynamic fee assets, the fees are paid from the active balance,
-		// and are not taken out of the redeem amount.
-		if isDynamicSwapper || tx.Confirmed {
-			effects.availableDiff[fromFeeAsset] -= int64(tx.Fees)
-		} else {
-			effects.pending[fromFeeAsset] -= tx.Fees
-		}
-		addFees(true, tx.Fees)
-		addDelta(true, int64(tx.Amount))
+		effects.fees[fromFeeAsset] += tx.Fees
```

Comment:
This is actually a small tweak to the accounting too. Fees are tracked separately now, so are not shown as a reduction in the pending, but they are still accounted for in the available balance and all of the final figures.